cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java

```diff
@@ -584,5 +585,54 @@ public static Literal createTypedLiteral(Model model, String literal, String dat
 //    return date != null ? model.createTypedLiteral(RdfUtils.format(date), XSD.dateTime.getURI()) : null;
 //  }
 
+
+  public static Model difference(Model a, Model b) {
+    Model result = a.difference(b);
+    Model actualResult = ModelFactory.createDefaultModel();
+
+    result.listStatements()
+          .forEach(statement -> {
+            RDFNode object = statement.getObject();
+
+            if (!object.isLiteral()) {
+              actualResult.add(statement);
+            }
+
+            Literal literal = object.asLiteral();
+            if (literal.getDatatype() == null) {
+              actualResult.add(statement);
+            }
+
+            if (!hasMatchingObject(b, statement)) {
```

Comment:
maybe add a comment why we do this, i know we discussed this, but other people will be clueless